### DISCUSSION TITLE
Bug: Traefik Configuration with Photoprism on Docker Desktop for Windows: permissions are too open

### DISCUSSION THREAD

**Discussion by HenningLa** [photoprism/photoprism#1997] [General]
First of all: thanks for making photoprism! Such great functionality, usability and design. It has fully exceeded my expectations as a substitute for google photos! Already set up a patreon subscription :)

Bug description:
I set up photoprism on a PC with windows 10 that I use as a server with docker desktop for windows. All was set up according to the photoprism getting started documentation.

I started running into issues when configuring traefik as reverse proxy based on the instructions in the photoprism docs: [Traefik - PhotoPrism](https://docs.photoprism.app/getting-started/proxies/traefik/). When accessing photoprism via https, I would only get a self-signed certificate, which would trigger browser warnings. The letsencrypt certificate would not be generated, with following log:

> The ACME resolver \"myresolver\" is skipped from the resolvers list because: unable to get ACME account: permissions 755 for /data/letsencrypt.json are too open, please use 600

Apparently, when sharing files from windows with the traefik container, [permissions are set to 755 and cannot be changed](https://stackoverflow.com/questions/66388158/changing-file-permissions-from-a-mounted-file-from-inside-the-docker-container) . This means, when using docker desktop for windows, a lot of fixes I found online did not work to change the permissions, for example using chmod inside the container via CLI or changing permissions with WSL via powershell.

I managed to fix it by creating the letsencrypt.json file only in the traefik container, instead of using a shared container. The according line in my docker compose file:

```
 traefik:
    volumes:
      - "./traefik.yaml:/etc/traefik/traefik.yaml"
      - "/data:/data"
```

Would this modification be worth a note in the photoprism traefik doc for users that use dockers for windows? It might be an obvious issue to fix for more experienced users. But I have to admit it took quite a while to fix, which might scare off other beginners that are setting up a first photoprism server on their windows machine? Is there any specific reason to create the letsencrypt.json file in a shared volume?

**Comment by lastzero** [comment]
You are right, only "root" or the user running Traefik should be able to read and write the "letsencrypt.json" file. We are aware that this can become difficult and complex when using different file and operating systems at the same time.

Feel free to send us a pull request with improvements to the documentation. In the long run we want to provide an automated solution that does not require advanced technical knowledge.

**Comment by abassonbhatoa** [comment]
@HenningLa 

Hi there

Thank you for posting, I have been have similar issues for weeks. I am trying to run Photoprism with Trarfik to expose to service to the internet so I can share albums. I too am trying to run these in Docker for Windows 10 and had tried changing the file permissions for letsencrypt.json.

I am a complete novice when it comes to Docker, reverse proxy, DNS and HTTPS in general and have never hosted before. I was just wondering if you wouldn't mind showing step by step how you managed to get yours working. Would I also need to setup up a domain name or could I just use my public IP address when I would like others to access it?

Many thanks!
